QuantConnect/Lean
# Historical data requests start & time fix

#### Description

Fix calculation `Time.GetStartTimeForTradeBars`, should round down in data timezone because the data timezone rounded down, converted into exchange time zone will be the datas Time.

![image](https://user-images.githubusercontent.com/3648114/93654512-8aa51f80-fa26-11ea-91f1-6d8cb01e5a01.png)



#### Related Issue




Closes #4035 

#### Motivation and Context

History start and end time are wrong, as a result we

#### Requires Documentation Change

N/A

#### How Has This Been Tested?



- manual testing
- add regression tests
- add unit tests for new implementation of `Time.GetStartTimeForTradeBars`

Fix existing test: `OpenInterestFutureUniverseSelectionModelTests.Can_Sort_By_Open_Interest` distory should return Sunday if market open

#### Types of changes


#### Checklist:
